File: Data Structures & Algorithms/duplicate-integer/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums) {
        let map = new Map();

        for(let i = 0; i < nums.length; i++){
            if(!map.get(nums[i])){
                map.set(nums[i], (map.get(nums[i]) || 0) + 1)
            } else {
                return true
            }
        }
        return false;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums:
            return 0

        nums.sort()

        max_res = 1
        i = 0

        while i < len(nums):
            j = i + 1
            res = 1

            while j < len(nums):
                if nums[j] == nums[j - 1]:
                    j += 1
                    i = j - 1
                    continue

                if nums[j] == nums[i] + 1:
                    res += 1
                    i = j
                    j += 1
                else:
                    break

            if res > max_res:
                max_res = res

            i += 1

        return max_res
